<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 12 Session information | Omics Data Analysis" />
<meta property="og:type" content="book" />

<meta property="og:description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2022-10-18" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain.">

<title>Chapter 12 Session information | Omics Data Analysis</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.25/datatables.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Omics Data Analysis<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-cli.html"><span class="toc-section-number">1</span> The shell</a>
<a href="sec-rmd.html"><span class="toc-section-number">2</span> R markdown</a>
<a href="sec-linmod.html"><span class="toc-section-number">3</span> Linear models</a>
<a href="sec-hts.html"><span class="toc-section-number">4</span> High throughput sequencing</a>
<a href="sec-rnaseq.html"><span class="toc-section-number">5</span> Differential expression analysis</a>
<a href="sec-design.html"><span class="toc-section-number">6</span> More complex experimental design</a>
<a href="sec-gsea.html"><span class="toc-section-number">7</span> Enrichment analyses</a>
<a href="sec-ms.html"><span class="toc-section-number">8</span> Mass spectrometry</a>
<a href="sec-prot.html"><span class="toc-section-number">9</span> Quantitative proteomics data analysis</a>
<a href="sec-ccl.html"><span class="toc-section-number">10</span> Conclusions</a>
<a href="sec-anx.html"><span class="toc-section-number">11</span> Annex</a>
<a id="active-page" href="sec-si.html"><span class="toc-section-number">12</span> Session information</a><ul class="toc-sections">
<li class="toc"><a href="#sec-setup2"> R package setup</a></li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><div id="sec-si" class="section level1" number="12">
<h1>
<span class="header-section-number">Chapter 12</span> Session information</h1>
<p>The following packages have been used to generate this document.</p>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="sec-si.html#cb470-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.1 (2022-06-23)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 22.04.1 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3
## LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] plotly_4.10.0               factoextra_1.0.7           
##  [3] gplots_3.1.3                limma_3.52.4               
##  [5] QFeatures_1.7.2             MultiAssayExperiment_1.22.0
##  [7] PSMatch_1.0.0               magrittr_2.0.3             
##  [9] Spectra_1.6.0               ProtGenerics_1.28.0        
## [11] BiocParallel_1.30.4         msigdbr_7.5.1              
## [13] clusterProfiler_4.4.4       GO.db_3.15.0               
## [15] org.Hs.eg.db_3.15.0         AnnotationDbi_1.58.0       
## [17] biomaRt_2.52.0              patchwork_1.1.2            
## [19] DESeq2_1.36.0               SummarizedExperiment_1.26.1
## [21] Biobase_2.56.0              MatrixGenerics_1.8.1       
## [23] matrixStats_0.62.0          GenomicRanges_1.48.0       
## [25] GenomeInfoDb_1.32.4         IRanges_2.30.1             
## [27] S4Vectors_0.34.0            BiocGenerics_0.42.0        
## [29] DT_0.25                     forcats_0.5.2              
## [31] stringr_1.4.1               dplyr_1.0.10               
## [33] purrr_0.3.5                 readr_2.1.3                
## [35] tidyr_1.2.1                 tibble_3.1.8               
## [37] tidyverse_1.3.2             ggplot2_3.3.6              
## [39] rWSBIM2122_0.4.1           
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2               shinydashboard_0.7.2     tidyselect_1.2.0        
##   [4] RSQLite_2.2.18           htmlwidgets_1.5.4        grid_4.2.1              
##   [7] scatterpie_0.1.8         munsell_0.5.0            preprocessCore_1.58.0   
##  [10] codetools_0.2-18         msdata_0.36.0            withr_2.5.0             
##  [13] colorspace_2.0-3         GOSemSim_2.22.0          filelock_1.0.2          
##  [16] highr_0.9                knitr_1.40               rstudioapi_0.14         
##  [19] ggsignif_0.6.4           DOSE_3.22.1              labeling_0.4.2          
##  [22] GenomeInfoDbData_1.2.8   polyclip_1.10-0          bit64_4.0.5             
##  [25] farver_2.1.1             downloader_0.4           vctrs_0.4.2             
##  [28] treeio_1.20.2            generics_0.1.3           xfun_0.33               
##  [31] BiocFileCache_2.4.0      R6_2.5.1                 clue_0.3-61             
##  [34] graphlayouts_0.8.2       locfit_1.5-9.6           MsCoreUtils_1.8.0       
##  [37] AnnotationFilter_1.20.0  bitops_1.0-7             cachem_1.0.6            
##  [40] fgsea_1.22.0             gridGraphics_0.5-1       DelayedArray_0.22.0     
##  [43] assertthat_0.2.1         promises_1.2.0.1         scales_1.2.1            
##  [46] vroom_1.6.0              ggraph_2.1.0             enrichplot_1.16.2       
##  [49] googlesheets4_1.0.1      gtable_0.3.1             tidygraph_1.2.2         
##  [52] rlang_1.0.6              genefilter_1.78.0        mzR_2.30.0              
##  [55] splines_4.2.1            rstatix_0.7.0            lazyeval_0.2.2          
##  [58] gargle_1.2.1             broom_1.0.1              abind_1.4-5             
##  [61] BiocManager_1.30.18      yaml_2.3.5               reshape2_1.4.4          
##  [64] modelr_0.1.9             crosstalk_1.2.0          backports_1.4.1         
##  [67] httpuv_1.6.6             qvalue_2.28.0            tools_4.2.1             
##  [70] bookdown_0.29            ggplotify_0.1.0          ellipsis_0.3.2          
##  [73] jquerylib_0.1.4          RColorBrewer_1.1-3       Rcpp_1.0.9              
##  [76] plyr_1.8.7               progress_1.2.2           zlibbioc_1.42.0         
##  [79] RCurl_1.98-1.9           prettyunits_1.1.1        ggpubr_0.4.0            
##  [82] viridis_0.6.2            cowplot_1.1.1            ExploreModelMatrix_1.8.0
##  [85] cluster_2.1.4            haven_2.5.1              ggrepel_0.9.1           
##  [88] fs_1.5.2                 data.table_1.14.2        DO.db_2.9               
##  [91] reprex_2.0.2             googledrive_2.0.0        msmbstyle_0.0.19        
##  [94] hms_1.1.2                shinyjs_2.1.0            mime_0.12               
##  [97] evaluate_0.17            xtable_1.8-4             XML_3.99-0.11           
## [100] readxl_1.4.1            
##  [ reached getOption("max.print") -- omitted 65 entries ]</code></pre>
<div id="sec-setup2" class="section level2" number="12.1">
<h2>
<span class="header-section-number">12.1</span> R package setup<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-setup2')" onmouseout="reset_tooltip('sec-setup2-tooltip')"><span class="tooltiptext" id="sec-setup2-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>To install all necessary packages to run all the code, please execute
the following code:</p>
<pre><code>pkgs &lt;- c("AnnotationDbi", "base", "Biobase", "BiocGenerics",
          "BiocParallel", "biomaRt", "clusterProfiler", "datasets", "DESeq2",
          "dplyr", "DT", "factoextra", "forcats", "GenomeInfoDb",
          "GenomicRanges", "ggplot2", "GO.db", "gplots", "graphics", "grDevices",
          "IRanges", "limma", "magrittr", "MatrixGenerics", "matrixStats",
          "methods", "msigdbr", "MultiAssayExperiment", "org.Hs.eg.db",
          "patchwork", "plotly", "ProtGenerics", "PSMatch", "purrr", "QFeatures",
          "readr", "rWSBIM2122", "S4Vectors", "Spectra", "stats", "stats4",
          "stringr", "SummarizedExperiment", "tibble", "tidyr", "tidyverse",
          "utils")</code></pre>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="sec-si.html#cb473-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>))</span>
<span id="cb473-2"><a href="sec-si.html#cb473-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb473-3"><a href="sec-si.html#cb473-3" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(pkgs)</span></code></pre></div>

</div>
</div></body></html>

<p style="text-align: center;">
<a href="sec-anx.html"><button class="btn btn-default">Previous</button></a>
</p>
<p class="build-date">Page built: 
2022-10-18
 using 
R version 4.2.1 (2022-06-23)
</p>
</div>
</div>



</body>
</html>
