<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 6 More complex experimental design | Omics Data Analysis" />
<meta property="og:type" content="book" />

<meta property="og:description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2023-08-26" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="/usr/share/javascript/mathjax/tex-mml-chtml.js" type="text/javascript"></script>

<meta name="description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain.">

<title>Chapter 6 More complex experimental design | Omics Data Analysis</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>
<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.10.2/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Omics Data Analysis<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-cli.html"><span class="toc-section-number">1</span> The shell</a>
<a href="sec-rmd.html"><span class="toc-section-number">2</span> R markdown</a>
<a href="sec-linmod.html"><span class="toc-section-number">3</span> Linear models</a>
<a href="sec-hts.html"><span class="toc-section-number">4</span> High throughput sequencing</a>
<a href="sec-rnaseq.html"><span class="toc-section-number">5</span> Differential expression analysis</a>
<a id="active-page" href="sec-design.html"><span class="toc-section-number">6</span> More complex experimental design</a><ul class="toc-sections">
<li class="toc"><a href="#a-more-complex-dataset"> A more complex dataset</a></li>
<li class="toc"><a href="#paired-design"> Paired design</a></li>
<li class="toc"><a href="#design-with-interaction"> Design with interaction</a></li>
</ul>
<a href="sec-gsea.html"><span class="toc-section-number">7</span> Enrichment analyses</a>
<a href="sec-ms.html"><span class="toc-section-number">8</span> Mass spectrometry</a>
<a href="sec-prot.html"><span class="toc-section-number">9</span> Quantitative proteomics data analysis</a>
<a href="sec-ccl.html"><span class="toc-section-number">10</span> Conclusions</a>
<a href="sec-anx.html"><span class="toc-section-number">11</span> Annex</a>
<a href="sec-si.html"><span class="toc-section-number">12</span> Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><div id="sec-design" class="section level1" number="6">
<h1>
<span class="header-section-number">Chapter 6</span> More complex experimental design</h1>
<p><strong>Learning Objectives</strong></p>
<p>The goal of this chapter is</p>
<ul>
<li>to understand more complex experimental designs</li>
<li>how to extract relevant factors</li>
</ul>
<div id="a-more-complex-dataset" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> A more complex dataset<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('a-more-complex-dataset')" onmouseout="reset_tooltip('a-more-complex-dataset-tooltip')"><span class="tooltiptext" id="a-more-complex-dataset-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>The dataset that we used previously was quite simple, we had 3 KD
samples vs 3 control samples. But the original dataset was in reality
a bit more complex, as the KD experiments were actually done on 2
different cell lines. So we have now 12 samples. The 6 first samples
correspond to an epithelial cell line (Cell1), and 6 other samples
correspond to a fibroblast cell line (Cell2).</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="sec-design.html#cb271-1" aria-hidden="true"></a><span class="kw">load</span>(<span class="st">"wsbim2122_data/deseq2/coldata_both.rda"</span>)</span>
<span id="cb271-2"><a href="sec-design.html#cb271-2" aria-hidden="true"></a><span class="kw">load</span>(<span class="st">"wsbim2122_data/deseq2/counts_both.rda"</span>)</span>
<span id="cb271-3"><a href="sec-design.html#cb271-3" aria-hidden="true"></a>coldata</span></code></pre></div>
<pre><code>##           Cell       Type Condition
## sample1  Cell1 Epithelial      mock
## sample2  Cell1 Epithelial      mock
## sample3  Cell1 Epithelial      mock
## sample4  Cell1 Epithelial        KD
## sample5  Cell1 Epithelial        KD
## sample6  Cell1 Epithelial        KD
## sample7  Cell2 Fibroblast      mock
## sample8  Cell2 Fibroblast      mock
## sample9  Cell2 Fibroblast      mock
## sample10 Cell2 Fibroblast        KD
## sample11 Cell2 Fibroblast        KD
## sample12 Cell2 Fibroblast        KD</code></pre>
<div id="pca-analysis" class="section level3" number="6.1.1">
<h3>
<span class="header-section-number">6.1.1</span> PCA analysis<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('pca-analysis')" onmouseout="reset_tooltip('pca-analysis-tooltip')"><span class="tooltiptext" id="pca-analysis-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Let’s start by a PCA analysis, that will help us to define the most
appropriate design to use for the analysis.</p>
<p>To generate the <a href="https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/DESeqDataSet-class">DESeqDataSet</a>, remember that we have to specify the experimental design.</p>
<p>We will see that can use different experimental designs for the analysis,
depending on the biological questions that we want to answer. We will discuss
about this latter. At this step, as we just want to do a PCA (that will anyway
be blind to the sample information specified by the design formula),
we can use a design ~ 1.</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="sec-design.html#cb273-1" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> counts,</span>
<span id="cb273-2"><a href="sec-design.html#cb273-2" aria-hidden="true"></a>                              <span class="dt">colData =</span> coldata,</span>
<span id="cb273-3"><a href="sec-design.html#cb273-3" aria-hidden="true"></a>                              <span class="dt">design =</span>  <span class="op">~</span><span class="st"> </span><span class="dv">1</span>)</span>
<span id="cb273-4"><a href="sec-design.html#cb273-4" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</span>
<span id="cb273-5"><a href="sec-design.html#cb273-5" aria-hidden="true"></a></span>
<span id="cb273-6"><a href="sec-design.html#cb273-6" aria-hidden="true"></a>rld &lt;-<span class="st"> </span><span class="kw">rlogTransformation</span>(dds)</span>
<span id="cb273-7"><a href="sec-design.html#cb273-7" aria-hidden="true"></a>pca_data &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(rld,</span>
<span id="cb273-8"><a href="sec-design.html#cb273-8" aria-hidden="true"></a>                    <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">"Condition"</span>, <span class="st">"Type"</span>),</span>
<span id="cb273-9"><a href="sec-design.html#cb273-9" aria-hidden="true"></a>                    <span class="dt">returnData =</span> <span class="ot">TRUE</span>)</span>
<span id="cb273-10"><a href="sec-design.html#cb273-10" aria-hidden="true"></a>percentVar &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="dv">100</span> <span class="op">*</span><span class="st"> </span><span class="kw">attr</span>(pca_data, <span class="st">"percentVar"</span>))</span>
<span id="cb273-11"><a href="sec-design.html#cb273-11" aria-hidden="true"></a><span class="kw">ggplot</span>(pca_data,</span>
<span id="cb273-12"><a href="sec-design.html#cb273-12" aria-hidden="true"></a>       <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2,</span>
<span id="cb273-13"><a href="sec-design.html#cb273-13" aria-hidden="true"></a>           <span class="dt">color =</span> Condition, <span class="dt">shape =</span> Type)) <span class="op">+</span></span>
<span id="cb273-14"><a href="sec-design.html#cb273-14" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>) <span class="op">+</span></span>
<span id="cb273-15"><a href="sec-design.html#cb273-15" aria-hidden="true"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="kw">paste0</span>(<span class="st">"PC1: "</span>,percentVar[<span class="dv">1</span>],<span class="st">"% variance"</span>)) <span class="op">+</span></span>
<span id="cb273-16"><a href="sec-design.html#cb273-16" aria-hidden="true"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="kw">paste0</span>(<span class="st">"PC2: "</span>,percentVar[<span class="dv">2</span>],<span class="st">"% variance"</span>))</span></code></pre></div>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-127-1.png" width="672"></p>
<p>We can see from this PCA plot that PC1 separates the samples based on the
cell line origin, while PC2 separates the samples based on the siRNA treatment.
This means that most of the variability comes from the cell origin.</p>
<p>The design used for the analysis will have to take this cell effect into
account.</p>
<p>Different designs can be used to analyse this dataset, depending on the
biological questions that are asked.</p>
<ul>
<li>
<strong>A paired design</strong> should be used to answer to the question:</li>
</ul>
<p><em>Which genes are consistently affected by the KD in both cell lines?</em></p>
<ul>
<li>
<strong>A design with an interaction</strong> should be used if the biological questions are:</li>
</ul>
<p><em>Which genes are significantly affected by the KD in the epithelial cells?</em></p>
<p><em>Which genes are significantly affected by the KD in the fibroblasts?</em></p>
<p><em>Which genes are not affected in the same way by the KD in the two cell lines</em></p>
</div>
</div>
<div id="paired-design" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> Paired design<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('paired-design')" onmouseout="reset_tooltip('paired-design-tooltip')"><span class="tooltiptext" id="paired-design-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>To take into account the condition effect while controlling the cell origin,
we can use the following design:</p>
<p><code>design = ~ Cell + Condition</code></p>
<p>Let’s run the analysis with this design (we have to re-run the
<code>DESeqDataSetFromMatrix()</code> and <code>DESeq()</code> functions)</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="sec-design.html#cb274-1" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> counts,</span>
<span id="cb274-2"><a href="sec-design.html#cb274-2" aria-hidden="true"></a>                              <span class="dt">colData =</span> coldata,</span>
<span id="cb274-3"><a href="sec-design.html#cb274-3" aria-hidden="true"></a>                              <span class="dt">design =</span>  <span class="op">~</span><span class="st"> </span>Type <span class="op">+</span><span class="st"> </span>Condition)</span>
<span id="cb274-4"><a href="sec-design.html#cb274-4" aria-hidden="true"></a></span>
<span id="cb274-5"><a href="sec-design.html#cb274-5" aria-hidden="true"></a><span class="co"># Set the mock level as the reference level</span></span>
<span id="cb274-6"><a href="sec-design.html#cb274-6" aria-hidden="true"></a>dds<span class="op">$</span>Condition &lt;-<span class="st"> </span><span class="kw">relevel</span>(dds<span class="op">$</span>Condition, <span class="dt">ref =</span> <span class="st">"mock"</span>)</span>
<span id="cb274-7"><a href="sec-design.html#cb274-7" aria-hidden="true"></a><span class="co"># Set the Epithelial cells as the reference level</span></span>
<span id="cb274-8"><a href="sec-design.html#cb274-8" aria-hidden="true"></a>dds<span class="op">$</span>Type &lt;-<span class="st"> </span><span class="kw">relevel</span>(dds<span class="op">$</span>Type, <span class="dt">ref =</span> <span class="st">"Epithelial"</span>)</span>
<span id="cb274-9"><a href="sec-design.html#cb274-9" aria-hidden="true"></a></span>
<span id="cb274-10"><a href="sec-design.html#cb274-10" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</span></code></pre></div>
<p>As we have seen previously, the <code>resultsNames()</code> function gives the
names of the coefficients that can be extracted.</p>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="sec-design.html#cb275-1" aria-hidden="true"></a><span class="kw">resultsNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] "Intercept"                     "Type_Fibroblast_vs_Epithelial"
## [3] "Condition_KD_vs_mock"</code></pre>
<p>The linear model set with this design can be written</p>
<p><span class="math display">\[log2(q_{ij}) = Intercept + Type\_Fibroblast\_vs\_Epithelial.x_j + Condition\_KD\_vs\_mock.y_j + \epsilon\]</span></p>
<ul>
<li><p>The <span class="math inline">\(Intercept\)</span> represents the log2 expression level in the
reference (in mock Epithelial cells).</p></li>
<li><p>The <span class="math inline">\(Type\_Fibroblast\_vs\_Epithelial\)</span> coefficient corresponds to the log2FC
between fibroblasts and epithelial cells. <span class="math inline">\(x_j = 0\)</span> if the sample j corresponds to epithelial cells,
and <span class="math inline">\(x_j = 1\)</span> if the sample j corresponds to fibroblasts.</p></li>
<li><p>The <span class="math inline">\(Condition\_KD\_vs\_mock\)</span> coefficient corresponds to the log2FC between KD and
mock cells. <span class="math inline">\(y_j = 0\)</span> if the sample j corresponds
to mock cells, and <span class="math inline">\(y_j\)</span> = 1 if the sample j corresponds to KD cells.</p></li>
</ul>
<p>The following figure (generated with the <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html">ExploreModelMatrix</a> package) illustrates the value of the
linear predictor of a generalized linear model for each combination of
input variables. It can help to understand and to generate contrasts.</p>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-130-1.png" width="672"></p>
<p>From this figure, we can see how the genes log2 expression values are modelized
in the different samples.</p>
<ul>
<li><p>In mock epithelial cells
<span class="math inline">\(log2(q_{Gene_i\_Epithelial\_mock}) = Intercept\)</span></p></li>
<li><p>In KD epithelial cells
<span class="math inline">\(log2(q_{Gene_i\_Epithelial\_KD}) = Intercept + Condition\_KD\_vs\_mock\)</span></p></li>
<li><p>In mock fibroblasts cells
<span class="math inline">\(log2(q_{Gene_i\_Fibroblast\_mock}) = Intercept + Type\_Fibroblast\_vs\_Epithelial\)</span></p></li>
<li><p>In KD fibroblasts cells
<span class="math inline">\(log2(q_{Gene_i\_Fibroblast\_KD}) = Intercept + Type\_Fibroblast\_vs\_Epithelial + Condition\_KD\_vs\_mock\)</span></p></li>
</ul>
<div id="kd-effect" class="section level3" number="6.2.1">
<h3>
<span class="header-section-number">6.2.1</span> KD effect<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('kd-effect')" onmouseout="reset_tooltip('kd-effect-tooltip')"><span class="tooltiptext" id="kd-effect-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>To identify the genes that are consistently affected by the KD in both cell lines,
we will have to test (for each gene) the <em>Condition_KD_vs_mock</em> coefficient, to see
if it is significantly different from zero.</p>
<p>Remember that we can use the <code>name</code> parameter of the <code>results()</code> function to specify the
coefficient that we want to extract (here the <em>Condition_KD_vs_mock</em> coefficient).
This will test if genes are differentially expressed in KD cells versus
mock cells.</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="sec-design.html#cb277-1" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">name =</span> <span class="st">"Condition_KD_vs_mock"</span>)</span>
<span id="cb277-2"><a href="sec-design.html#cb277-2" aria-hidden="true"></a>res_tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(res, <span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span></span>
<span id="cb277-3"><a href="sec-design.html#cb277-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">arrange</span>(padj)</span>
<span id="cb277-4"><a href="sec-design.html#cb277-4" aria-hidden="true"></a><span class="kw">head</span>(res_tbl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   ENSEMBL         baseMean log2FoldChange  lfcSE  stat   pvalue     padj
##   &lt;chr&gt;              &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 ENSG00000144674    2735.          1.11  0.0950  11.6 2.33e-31 3.87e-27
## 2 ENSG00000152700     967.          1.18  0.106   11.2 4.12e-29 3.03e-25
## 3 ENSG00000104763    3267.         -1.22  0.109  -11.2 5.46e-29 3.03e-25
## 4 ENSG00000122557     542.          0.945 0.0865  10.9 8.77e-28 3.65e-24
## 5 ENSG00000257103    2297.          1.37  0.130   10.5 9.08e-26 3.02e-22
## 6 ENSG00000116962    5115.          1.63  0.160   10.2 2.27e-24 6.29e-21</code></pre>
<p>The normalised counts of the genes with the lowest p-adjusted values are plotted bellow.
We can see that these genes correspond to genes that are highly affected by the siRNA
in both cells lines.</p>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="sec-design.html#cb279-1" aria-hidden="true"></a><span class="kw">as_tibble</span>(<span class="kw">counts</span>(dds[res_tbl<span class="op">$</span>ENSEMBL[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], ], <span class="dt">normalize =</span> <span class="ot">TRUE</span>),</span>
<span id="cb279-2"><a href="sec-design.html#cb279-2" aria-hidden="true"></a>          <span class="dt">rownames =</span> <span class="st">'ENSEMBL'</span>) <span class="op">%&gt;%</span></span>
<span id="cb279-3"><a href="sec-design.html#cb279-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">"sample"</span>, <span class="dt">values_to =</span> <span class="st">"counts"</span>, <span class="op">-</span>ENSEMBL) <span class="op">%&gt;%</span></span>
<span id="cb279-4"><a href="sec-design.html#cb279-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">as_tibble</span>(<span class="kw">colData</span>(dds), <span class="dt">rownames =</span> <span class="st">"sample"</span>)) <span class="op">%&gt;%</span></span>
<span id="cb279-5"><a href="sec-design.html#cb279-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">name =</span> <span class="kw">paste0</span>(<span class="kw">substr</span>(Type, <span class="dv">1</span>, <span class="dv">5</span>), <span class="st">'_'</span>, Condition, <span class="st">'_'</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)) <span class="op">%&gt;%</span></span>
<span id="cb279-6"><a href="sec-design.html#cb279-6" aria-hidden="true"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> name, <span class="dt">y =</span> counts, <span class="dt">fill =</span> Condition)) <span class="op">+</span></span>
<span id="cb279-7"><a href="sec-design.html#cb279-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">'identity'</span>, <span class="dt">color =</span> <span class="st">"gray30"</span>) <span class="op">+</span></span>
<span id="cb279-8"><a href="sec-design.html#cb279-8" aria-hidden="true"></a><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>ENSEMBL, <span class="dt">scales =</span> <span class="st">"free"</span>) <span class="op">+</span></span>
<span id="cb279-9"><a href="sec-design.html#cb279-9" aria-hidden="true"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb279-10"><a href="sec-design.html#cb279-10" aria-hidden="true"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb279-11"><a href="sec-design.html#cb279-11" aria-hidden="true"></a>        <span class="dt">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb279-12"><a href="sec-design.html#cb279-12" aria-hidden="true"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb279-13"><a href="sec-design.html#cb279-13" aria-hidden="true"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span></code></pre></div>
<pre><code>## Joining with `by = join_by(sample)`</code></pre>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-132-1.png" width="672"></p>
</div>
<div id="epithelial-vs-fibroblasts-cells" class="section level3" number="6.2.2">
<h3>
<span class="header-section-number">6.2.2</span> Epithelial vs fibroblasts cells<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('epithelial-vs-fibroblasts-cells')" onmouseout="reset_tooltip('epithelial-vs-fibroblasts-cells-tooltip')"><span class="tooltiptext" id="epithelial-vs-fibroblasts-cells-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Of course if the question of interest was</p>
<p><em>What are the genes that are differentially expressed in epithelial and fibroblasts?</em></p>
<p>we would have tested the <em>Type_Fibroblast_vs_Epithelial</em> coefficient.</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="sec-design.html#cb281-1" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">name =</span> <span class="st">"Type_Fibroblast_vs_Epithelial"</span>)</span>
<span id="cb281-2"><a href="sec-design.html#cb281-2" aria-hidden="true"></a>res_tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(res, <span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span></span>
<span id="cb281-3"><a href="sec-design.html#cb281-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">arrange</span>(padj)</span>
<span id="cb281-4"><a href="sec-design.html#cb281-4" aria-hidden="true"></a><span class="kw">as_tibble</span>(<span class="kw">counts</span>(dds[res_tbl<span class="op">$</span>ENSEMBL[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], ], <span class="dt">normalize =</span> <span class="ot">TRUE</span>),</span>
<span id="cb281-5"><a href="sec-design.html#cb281-5" aria-hidden="true"></a>          <span class="dt">rownames =</span> <span class="st">'ENSEMBL'</span>) <span class="op">%&gt;%</span></span>
<span id="cb281-6"><a href="sec-design.html#cb281-6" aria-hidden="true"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">"sample"</span>, <span class="dt">values_to =</span> <span class="st">"counts"</span>, <span class="op">-</span>ENSEMBL) <span class="op">%&gt;%</span></span>
<span id="cb281-7"><a href="sec-design.html#cb281-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">as_tibble</span>(<span class="kw">colData</span>(dds), <span class="dt">rownames =</span> <span class="st">"sample"</span>)) <span class="op">%&gt;%</span></span>
<span id="cb281-8"><a href="sec-design.html#cb281-8" aria-hidden="true"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">name =</span> <span class="kw">paste0</span>(<span class="kw">substr</span>(Type, <span class="dv">1</span>, <span class="dv">5</span>), <span class="st">'_'</span>, Condition, <span class="st">'_'</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)) <span class="op">%&gt;%</span></span>
<span id="cb281-9"><a href="sec-design.html#cb281-9" aria-hidden="true"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> name, <span class="dt">y =</span> counts, <span class="dt">fill =</span> Condition)) <span class="op">+</span></span>
<span id="cb281-10"><a href="sec-design.html#cb281-10" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">'identity'</span>, <span class="dt">color =</span> <span class="st">"gray30"</span>) <span class="op">+</span></span>
<span id="cb281-11"><a href="sec-design.html#cb281-11" aria-hidden="true"></a><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>ENSEMBL, <span class="dt">scales =</span> <span class="st">"free"</span>) <span class="op">+</span></span>
<span id="cb281-12"><a href="sec-design.html#cb281-12" aria-hidden="true"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb281-13"><a href="sec-design.html#cb281-13" aria-hidden="true"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb281-14"><a href="sec-design.html#cb281-14" aria-hidden="true"></a>        <span class="dt">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb281-15"><a href="sec-design.html#cb281-15" aria-hidden="true"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb281-16"><a href="sec-design.html#cb281-16" aria-hidden="true"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span></code></pre></div>
<pre><code>## Joining with `by = join_by(sample)`</code></pre>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-133-1.png" width="672"></p>
</div>
</div>
<div id="design-with-interaction" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> Design with interaction<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('design-with-interaction')" onmouseout="reset_tooltip('design-with-interaction-tooltip')"><span class="tooltiptext" id="design-with-interaction-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Interaction terms can be added to the design formula, in order to test
the treatment effect in one cell line or the treatment
effect that differs across the cell lines.</p>
<p>The design with an interaction could be written <span class="math display">\[design = ~ Cell * Condition\]</span>
which is equivalent to
<span class="math display">\[design = ~ Cell + Condition + Cell:Condition\]</span></p>
<p>Let’s use this design in our analysis (we have to re-run the
<code>DESeqDataSetFromMatrix()</code> and <code>DESeq()</code> functions)</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="sec-design.html#cb283-1" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> counts,</span>
<span id="cb283-2"><a href="sec-design.html#cb283-2" aria-hidden="true"></a>                              <span class="dt">colData =</span> coldata,</span>
<span id="cb283-3"><a href="sec-design.html#cb283-3" aria-hidden="true"></a>                              <span class="dt">design =</span>  <span class="op">~</span><span class="st"> </span>Type <span class="op">*</span><span class="st"> </span>Condition)</span>
<span id="cb283-4"><a href="sec-design.html#cb283-4" aria-hidden="true"></a><span class="co"># Set the Epithelial cells as the reference level</span></span>
<span id="cb283-5"><a href="sec-design.html#cb283-5" aria-hidden="true"></a>dds<span class="op">$</span>Type &lt;-<span class="st"> </span><span class="kw">relevel</span>(dds<span class="op">$</span>Type, <span class="dt">ref =</span> <span class="st">"Epithelial"</span>)</span>
<span id="cb283-6"><a href="sec-design.html#cb283-6" aria-hidden="true"></a><span class="co"># Set the mock condition as the reference level</span></span>
<span id="cb283-7"><a href="sec-design.html#cb283-7" aria-hidden="true"></a>dds<span class="op">$</span>Condition &lt;-<span class="st"> </span><span class="kw">relevel</span>(dds<span class="op">$</span>Condition, <span class="dt">ref =</span> <span class="st">"mock"</span>)</span>
<span id="cb283-8"><a href="sec-design.html#cb283-8" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</span></code></pre></div>
<p>The <code>resultsNames()</code> function gives the names of results that can be extracted.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="sec-design.html#cb284-1" aria-hidden="true"></a><span class="kw">resultsNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] "Intercept"                     "Type_Fibroblast_vs_Epithelial"
## [3] "Condition_KD_vs_mock"          "TypeFibroblast.ConditionKD"</code></pre>
<p>The linear model used with this design can be written</p>
<p><span class="math display">\[log2(q_{ij}) = Intercept + Type\_Fibroblast\_vs\_Epithelial.x_j + Condition\_KD\_vs\_mock.y_j  + TypeFibroblast.ConditionKD.z_j + \epsilon\]</span></p>
<ul>
<li><p>The <span class="math inline">\(Intercept\)</span> represents the log2 expression level in the
reference (in mock Cell1).</p></li>
<li><p>The <span class="math inline">\(Type\_Fibroblast\_vs\_Epithelial\)</span> coefficient corresponds
to the log2FC between mock fibroblasts and mock epithelial cells.
<span class="math inline">\(x_j = 0\)</span> if the sample j corresponds to fibroblasts,
and <span class="math inline">\(x_j = 1\)</span> if the sample j corresponds to epithelial cells.</p></li>
<li><p>The <span class="math inline">\(Condition\_KD\_vs\_mock\)</span> coefficient corresponds to the log2FC
between KD and mock cells in the reference cells (here epithelial cells).
<span class="math inline">\(y_j = 0\)</span> if the sample j corresponds to mock cells, and <span class="math inline">\(y_j\)</span> = 1 if
the sample j corresponds to KD cells.</p></li>
<li><p>The <span class="math inline">\(TypeFibroblast.ConditionKD\)</span> coefficient corresponds to the
the <em>extra KD effect</em> in KD-fibroblasts compared to KD-epitelial cells.
<span class="math inline">\(z_j = 0\)</span> if the sample j corresponds to mock cells or to epithial cells,
and <span class="math inline">\(z_j = 1\)</span> if the sample j corresponds to KD-fibroblasts.</p></li>
</ul>
<p>This is summarized in the following figure:</p>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-136-1.png" width="672"></p>
<p>From this figure, we can see how the genes log2 expression values
are modelized in the different samples.</p>
<ul>
<li><p>in mock epithelial cells
<span class="math inline">\(log2(q_{Gene_i\_Epithelial\_mock}) = Intercept\)</span></p></li>
<li><p>in KD epithelial cells
<span class="math inline">\(log2(q_{Gene_i\_Epithelial\_KD}) = Intercept + Condition\_KD\_vs\_mock\)</span></p></li>
<li><p>in mock fibroblasts cells
<span class="math inline">\(log2(q_{Gene_i\_Fibroblast\_mock}) = Intercept + Type\_Fibroblast\_vs\_Epithelial\)</span></p></li>
<li><p>in KD fibroblasts cells
<span class="math inline">\(log2(q_{Gene_i\_Fibroblast\_KD}) = Intercept + Type\_Fibroblast\_vs\_Epithelial + Condition\_KD\_vs\_mock + TypeFibroblast.ConditionKD\)</span></p></li>
</ul>
<div id="kd-effect-in-epithelial-cells" class="section level3" number="6.3.1">
<h3>
<span class="header-section-number">6.3.1</span> KD effect in Epithelial cells<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('kd-effect-in-epithelial-cells')" onmouseout="reset_tooltip('kd-effect-in-epithelial-cells-tooltip')"><span class="tooltiptext" id="kd-effect-in-epithelial-cells-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>As the epithelial cells were set as the reference cells,
the <em>Condition_KD_vs_mock</em> coefficient should be extracted to
test the effect of the KD in these cells.</p>
<p>In fact, testing the siRNA effect in epithelial cells is like comparing the
logFC between epithelial KD and epithelial mock cells.</p>
<p><span class="math display">\[log2(\frac{q_{Gene_i\_Epithelial\_KD}}{q_{Gene_i\_Epithelial\_mock}})\]</span></p>
<p><span class="math display">\[= log2(q_{Gene_i\_Epithelial\_KD}) - log2(q_{Gene_i\_Epithelial\_mock})\]</span>
<span class="math display">\[= Intercept + Condition\_KD - Intercept = Condition\_KD\]</span></p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="sec-design.html#cb286-1" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">name =</span> <span class="st">"Condition_KD_vs_mock"</span>)</span>
<span id="cb286-2"><a href="sec-design.html#cb286-2" aria-hidden="true"></a>res_tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(res, <span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span></span>
<span id="cb286-3"><a href="sec-design.html#cb286-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">arrange</span>(padj)</span>
<span id="cb286-4"><a href="sec-design.html#cb286-4" aria-hidden="true"></a><span class="kw">head</span>(res_tbl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   ENSEMBL         baseMean log2FoldChange lfcSE  stat   pvalue     padj
##   &lt;chr&gt;              &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 ENSG00000101856    1535.          -4.66 0.253 -18.4 1.76e-75 3.17e-71
## 2 ENSG00000177494     133.           6.26 0.366  17.1 1.09e-65 9.77e-62
## 3 ENSG00000008513    1198.           3.02 0.223  13.6 7.08e-42 4.25e-38
## 4 ENSG00000175315     106.           4.30 0.329  13.1 5.74e-39 2.58e-35
## 5 ENSG00000128245    2877.           2.07 0.166  12.4 1.76e-35 6.33e-32
## 6 ENSG00000115107     579.           3.46 0.279  12.4 3.31e-35 9.92e-32</code></pre>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="sec-design.html#cb288-1" aria-hidden="true"></a><span class="kw">as_tibble</span>(<span class="kw">counts</span>(dds[res_tbl<span class="op">$</span>ENSEMBL[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], ], <span class="dt">normalize =</span> <span class="ot">TRUE</span>),</span>
<span id="cb288-2"><a href="sec-design.html#cb288-2" aria-hidden="true"></a>          <span class="dt">rownames =</span> <span class="st">'ENSEMBL'</span>) <span class="op">%&gt;%</span></span>
<span id="cb288-3"><a href="sec-design.html#cb288-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">"sample"</span>, <span class="dt">values_to =</span> <span class="st">"counts"</span>, <span class="op">-</span>ENSEMBL) <span class="op">%&gt;%</span></span>
<span id="cb288-4"><a href="sec-design.html#cb288-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">as_tibble</span>(<span class="kw">colData</span>(dds), <span class="dt">rownames =</span> <span class="st">"sample"</span>)) <span class="op">%&gt;%</span></span>
<span id="cb288-5"><a href="sec-design.html#cb288-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">name =</span> <span class="kw">paste0</span>(<span class="kw">substr</span>(Type, <span class="dv">1</span>, <span class="dv">5</span>), <span class="st">'_'</span>, Condition, <span class="st">'_'</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)) <span class="op">%&gt;%</span></span>
<span id="cb288-6"><a href="sec-design.html#cb288-6" aria-hidden="true"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> name, <span class="dt">y =</span> counts, <span class="dt">fill =</span> Condition)) <span class="op">+</span></span>
<span id="cb288-7"><a href="sec-design.html#cb288-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">'identity'</span>, <span class="dt">color =</span> <span class="st">"gray30"</span>) <span class="op">+</span></span>
<span id="cb288-8"><a href="sec-design.html#cb288-8" aria-hidden="true"></a><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>ENSEMBL, <span class="dt">scales =</span> <span class="st">"free"</span>) <span class="op">+</span></span>
<span id="cb288-9"><a href="sec-design.html#cb288-9" aria-hidden="true"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb288-10"><a href="sec-design.html#cb288-10" aria-hidden="true"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb288-11"><a href="sec-design.html#cb288-11" aria-hidden="true"></a>        <span class="dt">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb288-12"><a href="sec-design.html#cb288-12" aria-hidden="true"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb288-13"><a href="sec-design.html#cb288-13" aria-hidden="true"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span></code></pre></div>
<pre><code>## Joining with `by = join_by(sample)`</code></pre>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-137-1.png" width="672"></p>
</div>
<div id="kd-effect-in-fibroblasts-cells" class="section level3" number="6.3.2">
<h3>
<span class="header-section-number">6.3.2</span> KD effect in Fibroblasts cells<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('kd-effect-in-fibroblasts-cells')" onmouseout="reset_tooltip('kd-effect-in-fibroblasts-cells-tooltip')"><span class="tooltiptext" id="kd-effect-in-fibroblasts-cells-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>As the fibroblasts cells were NOT set as the reference cells,
it is less trivial to extract the right coefficients to test the
KD effect in fibroblasts.</p>
<p>But remember that testing the siRNA effect in fibroblast cells is like
comparing the logFC between fibroblast KD and fibroblast mock cells.</p>
<p><span class="math display">\[log2(\frac{q_{Gene_i\_Fibroblast\_KD}}{q_{Gene_i\_Fibroblast\_mock}})\]</span></p>
<p><span class="math display">\[= log2(q_{Gene_i\_Fibroblast\_KD}) - log2(q_{Gene_i\_Fibroblast\_mock})\]</span>
<span class="math display">\[= (Intercept + Condition\_KD + Type\_Fibroblast\_vs\_Epithelial +  TypeFibroblast.ConditionKD )\]</span> <span class="math display">\[ - (Intercept + Type\_Fibroblast\_vs\_Epithelial) \]</span>
<span class="math display">\[= Condition\_KD +  TypeFibroblast.ConditionKD \]</span></p>
<p>So to test the KD effect in fibroblasts, we have to extract both
<em>Condition_KD_vs_mock</em> and <em>TypeFibroblast.ConditionKD</em> coefficients.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="sec-design.html#cb290-1" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds,</span>
<span id="cb290-2"><a href="sec-design.html#cb290-2" aria-hidden="true"></a>               <span class="kw">list</span>(<span class="kw">c</span>(<span class="st">"Condition_KD_vs_mock"</span>, <span class="st">"TypeFibroblast.ConditionKD"</span>)))</span>
<span id="cb290-3"><a href="sec-design.html#cb290-3" aria-hidden="true"></a>res_tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(res, <span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span></span>
<span id="cb290-4"><a href="sec-design.html#cb290-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">arrange</span>(padj)</span>
<span id="cb290-5"><a href="sec-design.html#cb290-5" aria-hidden="true"></a><span class="kw">head</span>(res_tbl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   ENSEMBL         baseMean log2FoldChange lfcSE   stat   pvalue     padj
##   &lt;chr&gt;              &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 ENSG00000151632     500.          -2.99 0.291 -10.3  1.12e-24 1.55e-20
## 2 ENSG00000271303     409.          -2.14 0.232  -9.23 2.60e-20 1.80e-16
## 3 ENSG00000138678     366.          -1.86 0.207  -8.99 2.51e-19 1.16e-15
## 4 ENSG00000101856    1535.          -1.99 0.240  -8.28 1.19e-16 4.14e-13
## 5 ENSG00000136167     150.          -2.88 0.354  -8.15 3.73e-16 1.04e-12
## 6 ENSG00000144674    2735.           1.04 0.129   8.05 8.22e-16 1.63e-12</code></pre>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="sec-design.html#cb292-1" aria-hidden="true"></a><span class="kw">as_tibble</span>(<span class="kw">counts</span>(dds[res_tbl<span class="op">$</span>ENSEMBL[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], ], <span class="dt">normalize =</span> <span class="ot">TRUE</span>),</span>
<span id="cb292-2"><a href="sec-design.html#cb292-2" aria-hidden="true"></a>          <span class="dt">rownames =</span> <span class="st">'ENSEMBL'</span>) <span class="op">%&gt;%</span></span>
<span id="cb292-3"><a href="sec-design.html#cb292-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">"sample"</span>, <span class="dt">values_to =</span> <span class="st">"counts"</span>, <span class="op">-</span>ENSEMBL) <span class="op">%&gt;%</span></span>
<span id="cb292-4"><a href="sec-design.html#cb292-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">as_tibble</span>(<span class="kw">colData</span>(dds), <span class="dt">rownames =</span> <span class="st">"sample"</span>)) <span class="op">%&gt;%</span></span>
<span id="cb292-5"><a href="sec-design.html#cb292-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">name =</span> <span class="kw">paste0</span>(<span class="kw">substr</span>(Type, <span class="dv">1</span>, <span class="dv">5</span>), <span class="st">'_'</span>, Condition, <span class="st">'_'</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)) <span class="op">%&gt;%</span></span>
<span id="cb292-6"><a href="sec-design.html#cb292-6" aria-hidden="true"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> name, <span class="dt">y =</span> counts, <span class="dt">fill =</span> Condition)) <span class="op">+</span></span>
<span id="cb292-7"><a href="sec-design.html#cb292-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">'identity'</span>, <span class="dt">color =</span> <span class="st">"gray30"</span>) <span class="op">+</span></span>
<span id="cb292-8"><a href="sec-design.html#cb292-8" aria-hidden="true"></a><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>ENSEMBL, <span class="dt">scales =</span> <span class="st">"free"</span>) <span class="op">+</span></span>
<span id="cb292-9"><a href="sec-design.html#cb292-9" aria-hidden="true"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb292-10"><a href="sec-design.html#cb292-10" aria-hidden="true"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb292-11"><a href="sec-design.html#cb292-11" aria-hidden="true"></a>        <span class="dt">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb292-12"><a href="sec-design.html#cb292-12" aria-hidden="true"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb292-13"><a href="sec-design.html#cb292-13" aria-hidden="true"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span></code></pre></div>
<pre><code>## Joining with `by = join_by(sample)`</code></pre>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-138-1.png" width="672"></p>
<p>Note that another possibility would have been to set the fibroblasts
as the reference cells, re-run the whole analysis, and to simply
extract the <em>Condition_KD_vs_mock</em> coefficient.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="sec-design.html#cb294-1" aria-hidden="true"></a>dds_test &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> counts,</span>
<span id="cb294-2"><a href="sec-design.html#cb294-2" aria-hidden="true"></a>                              <span class="dt">colData =</span> coldata,</span>
<span id="cb294-3"><a href="sec-design.html#cb294-3" aria-hidden="true"></a>                              <span class="dt">design =</span>  <span class="op">~</span><span class="st"> </span>Type <span class="op">*</span><span class="st"> </span>Condition)</span>
<span id="cb294-4"><a href="sec-design.html#cb294-4" aria-hidden="true"></a><span class="co"># Set the Epithelial cells as the reference level</span></span>
<span id="cb294-5"><a href="sec-design.html#cb294-5" aria-hidden="true"></a>dds_test<span class="op">$</span>Type &lt;-<span class="st"> </span><span class="kw">relevel</span>(dds_test<span class="op">$</span>Type, <span class="dt">ref =</span> <span class="st">"Fibroblast"</span>)</span>
<span id="cb294-6"><a href="sec-design.html#cb294-6" aria-hidden="true"></a><span class="co"># Set the mock condition as the reference level</span></span>
<span id="cb294-7"><a href="sec-design.html#cb294-7" aria-hidden="true"></a>dds_test<span class="op">$</span>Condition &lt;-<span class="st"> </span><span class="kw">relevel</span>(dds_test<span class="op">$</span>Condition, <span class="dt">ref =</span> <span class="st">"mock"</span>)</span>
<span id="cb294-8"><a href="sec-design.html#cb294-8" aria-hidden="true"></a>dds_test &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds_test)</span>
<span id="cb294-9"><a href="sec-design.html#cb294-9" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds_test,</span>
<span id="cb294-10"><a href="sec-design.html#cb294-10" aria-hidden="true"></a>               <span class="dt">name =</span> <span class="st">"Condition_KD_vs_mock"</span>)</span>
<span id="cb294-11"><a href="sec-design.html#cb294-11" aria-hidden="true"></a>res_tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(res, <span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span></span>
<span id="cb294-12"><a href="sec-design.html#cb294-12" aria-hidden="true"></a><span class="st">  </span><span class="kw">arrange</span>(padj)</span>
<span id="cb294-13"><a href="sec-design.html#cb294-13" aria-hidden="true"></a><span class="kw">head</span>(res_tbl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   ENSEMBL         baseMean log2FoldChange lfcSE   stat   pvalue     padj
##   &lt;chr&gt;              &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 ENSG00000151632     500.          -2.99 0.291 -10.3  1.12e-24 1.55e-20
## 2 ENSG00000271303     409.          -2.14 0.232  -9.23 2.60e-20 1.80e-16
## 3 ENSG00000138678     366.          -1.86 0.207  -8.99 2.51e-19 1.16e-15
## 4 ENSG00000101856    1535.          -1.99 0.240  -8.28 1.19e-16 4.14e-13
## 5 ENSG00000136167     150.          -2.88 0.354  -8.15 3.73e-16 1.04e-12
## 6 ENSG00000144674    2735.           1.04 0.129   8.05 8.22e-16 1.63e-12</code></pre>
</div>
<div id="kd-effect-that-are-different-across-epithelial-and-fibroblasts-cells" class="section level3" number="6.3.3">
<h3>
<span class="header-section-number">6.3.3</span> KD effect that are different across epithelial and fibroblasts cells<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('kd-effect-that-are-different-across-epithelial-and-fibroblasts-cells')" onmouseout="reset_tooltip('kd-effect-that-are-different-across-epithelial-and-fibroblasts-cells-tooltip')"><span class="tooltiptext" id="kd-effect-that-are-different-across-epithelial-and-fibroblasts-cells-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Which genes are not affected in the same way by the KD in the two cell lines?
In other terms, we are testing the KD effects that are significantly different in both cell
lines. This can be done by extracting the <em>TypeFibroblast.ConditionKD</em> coefficient.
In fact this coefficient corresponds to the <em>extra KD effect</em> ocuring in fibroblasts
compared to epitelial cells.</p>
<p>If the effect of the KD is similar in both cell types, we expect the
<em>TypeFibroblast.ConditionKD</em> coefficient to be nearly 0.</p>
<p>If the effect of the KD is higher in fibroblasts than in epithelial cells, we expect the
<em>TypeFibroblast.ConditionKD</em> coefficient to be higher than 0.</p>
<p>If the effect of the KD is lower in fibroblasts than in epithelial cells, we expect the
<em>TypeFibroblast.ConditionKD</em> coefficient to be lower than 0.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="sec-design.html#cb296-1" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">name =</span> <span class="st">"TypeFibroblast.ConditionKD"</span>)</span>
<span id="cb296-2"><a href="sec-design.html#cb296-2" aria-hidden="true"></a>res_tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(res, <span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span></span>
<span id="cb296-3"><a href="sec-design.html#cb296-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">arrange</span>(padj)</span>
<span id="cb296-4"><a href="sec-design.html#cb296-4" aria-hidden="true"></a><span class="kw">head</span>(res_tbl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   ENSEMBL         baseMean log2FoldChange lfcSE   stat   pvalue     padj
##   &lt;chr&gt;              &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 ENSG00000177494     133.          -6.76 0.670 -10.1  5.97e-24 8.70e-20
## 2 ENSG00000198846     238.          -2.95 0.297  -9.93 3.04e-23 2.21e-19
## 3 ENSG00000100097   20864.           1.93 0.196   9.84 7.78e-23 3.78e-19
## 4 ENSG00000171132     335.          -2.33 0.255  -9.12 7.49e-20 2.73e-16
## 5 ENSG00000108179    2994.          -1.83 0.201  -9.08 1.07e-19 3.13e-16
## 6 ENSG00000008513    1198.          -2.77 0.313  -8.86 7.85e-19 1.91e-15</code></pre>
<p>We can see that gene ENSG00000177494 (with the lowest p-value) has as
highly negative log2FC. This means that the effect of the siRNA treatment
is much lower in fibroblasts than in epithelial cells.</p>
<p>Gene ENSG00000108179 is also an illustrative example,
as its value is decreased by the siRNA treatment in the epithelial cells,
while on the contrary it is increased by the treatment in the fibroblast cells.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="sec-design.html#cb298-1" aria-hidden="true"></a><span class="kw">as_tibble</span>(<span class="kw">counts</span>(dds[res_tbl<span class="op">$</span>ENSEMBL[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">5</span>)], ], <span class="dt">normalize =</span> <span class="ot">TRUE</span>),</span>
<span id="cb298-2"><a href="sec-design.html#cb298-2" aria-hidden="true"></a>          <span class="dt">rownames =</span> <span class="st">'ENSEMBL'</span>) <span class="op">%&gt;%</span></span>
<span id="cb298-3"><a href="sec-design.html#cb298-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">"sample"</span>, <span class="dt">values_to =</span> <span class="st">"counts"</span>, <span class="op">-</span>ENSEMBL) <span class="op">%&gt;%</span></span>
<span id="cb298-4"><a href="sec-design.html#cb298-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">as_tibble</span>(<span class="kw">colData</span>(dds), <span class="dt">rownames =</span> <span class="st">"sample"</span>)) <span class="op">%&gt;%</span></span>
<span id="cb298-5"><a href="sec-design.html#cb298-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">name =</span> <span class="kw">paste0</span>(<span class="kw">substr</span>(Type, <span class="dv">1</span>, <span class="dv">5</span>), <span class="st">'_'</span>, Condition, <span class="st">'_'</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)) <span class="op">%&gt;%</span></span>
<span id="cb298-6"><a href="sec-design.html#cb298-6" aria-hidden="true"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> name, <span class="dt">y =</span> counts, <span class="dt">fill =</span> Condition)) <span class="op">+</span></span>
<span id="cb298-7"><a href="sec-design.html#cb298-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">'identity'</span>, <span class="dt">color =</span> <span class="st">"gray30"</span>) <span class="op">+</span></span>
<span id="cb298-8"><a href="sec-design.html#cb298-8" aria-hidden="true"></a><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>ENSEMBL, <span class="dt">scales =</span> <span class="st">"free"</span>) <span class="op">+</span></span>
<span id="cb298-9"><a href="sec-design.html#cb298-9" aria-hidden="true"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">8</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb298-10"><a href="sec-design.html#cb298-10" aria-hidden="true"></a>        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb298-11"><a href="sec-design.html#cb298-11" aria-hidden="true"></a>        <span class="dt">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb298-12"><a href="sec-design.html#cb298-12" aria-hidden="true"></a>        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>),</span>
<span id="cb298-13"><a href="sec-design.html#cb298-13" aria-hidden="true"></a>        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">7</span>))</span></code></pre></div>
<pre><code>## Joining with `by = join_by(sample)`</code></pre>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-141-1.png" width="672"></p>

</div>
</div>
</div></body></html>

<p style="text-align: center;">
<a href="sec-rnaseq.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-gsea.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2023-08-26
 using 
R version 4.3.1 (2023-06-16)
</p>
</div>
</div>



</body>
</html>
